django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -64,92 +61,110 @@ class PackageListResponseSerializer(serializers.Serializer):
     results = CyberstormPackagePreviewSerializer(many=True)
 
 
-class BasePackageListApiView(ListAPIView):
+class PackageListPaginator(PageNumberPagination):
+    page_size = 20
+
+    def get_schema_fields(self, view):
+        """
+        By default this would return page (via page_query_param
+        attribute) and page_size (via page_size_query_param attribute).
+        The former would clash with the page field defined in our
+        PackageListRequestSerializer, breaking the API documentation
+        generated by drf_yasg. Returning an empty list means we have the
+        control and responsibility to make sure the documentation
+        matches reality.
+        """
+        return []
+
+
+class BasePackageListAPIView(ListAPIView):
     """
     Base class for community-scoped, paginated, filterable package listings.
 
     Classes implementing this base class should receive `community_id`
-    url parameter and implement the abstract methods listed below.
+    url parameter.
+
+    Methods with names starting prefixed with underscore are you custom
+    methods, whereas the rest are overwritten methods from ListAPIView.
     """
 
-    page_size = 20
+    pagination_class = PackageListPaginator
+    serializer_class = CyberstormPackagePreviewSerializer
+    viewname: str = ""  # Define in subclass
 
-    @abstractmethod
-    def _get_paginator_cache_key(self) -> str:
-        """
-        Return cache_key argument for CachedPaginator initialization.
-        """
-        raise NotImplementedError(
-            "PackageListApiView must implement _get_paginator_cache_key",
-        )
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
```

Comment:
Seems like this could be removed entirely?

Not a blocker

Reply:
Yeah, it's a leftover from debugging the page parameter issue.